homeassistant/components/luxtronik/sensor.py

```diff
@@ -0,0 +1,187 @@
+"""
+Support for monitoring the state of a Luxtronik heatpump.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/integrations/luxtronik/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_FRIENDLY_NAME,
+    CONF_ICON,
+    CONF_ID,
+    CONF_SENSORS,
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_TIMESTAMP,
+    ENERGY_KILO_WATT_HOUR,
+    PRESSURE_BAR,
+    TEMP_CELSIUS,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import slugify
+
+from . import DATA_LUXTRONIK, ENTITY_ID_FORMAT
+from .const import (
+    CONF_BAR,
+    CONF_CALCULATIONS,
+    CONF_CELSIUS,
+    CONF_ENERGY,
+    CONF_FLOW,
+    CONF_GROUP,
+    CONF_HOURS,
+    CONF_KELVIN,
+    CONF_PARAMETERS,
+    CONF_PERCENT,
+    CONF_SECONDS,
+    CONF_TIMESTAMP,
+    CONF_VISIBILITIES,
+    CONF_VOLTAGE,
+    FLOW_LITERS_PER_MINUTE,
+    PERCENTAGE_PERCENT,
+    TEMP_KELVIN,
+    TIME_HOUR,
+    TIME_SECONDS,
+    VOLTAGE_VOLT,
+)
+
+ICONS = {
+    "celsius": "mdi:thermometer",
+    "seconds": "mdi:timer-sand",
+    "pulses": "mdi:pulse",
+    "ipaddress": "mdi:ip-network-outline",
+    "timestamp": "mdi:calendar-range",
+    "errorcode": "mdi:alert-circle-outline",
+    "kelvin": "mdi:thermometer",
+    "bar": "mdi:arrow-collapse-all",
+    "percent": "mdi:percent",
+    "rpm": "mdi:rotate-right",
+    "energy": "mdi:flash-circle",
+    "voltage": "mdi:flash-outline",
+    "hours": "mdi:clock-outline",
+    "flow": "mdi:chart-bell-curve",
+    "level": "mdi:format-list-numbered",
+    "count": "mdi:counter",
+    "version": "mdi:information-outline",
+}
+
+DEVICE_CLASSES = {
+    CONF_CELSIUS: DEVICE_CLASS_PRESSURE,
+    CONF_KELVIN: DEVICE_CLASS_PRESSURE,
+    CONF_BAR: DEVICE_CLASS_PRESSURE,
+    CONF_SECONDS: DEVICE_CLASS_TIMESTAMP,
+    CONF_HOURS: DEVICE_CLASS_TIMESTAMP,
+    CONF_TIMESTAMP: DEVICE_CLASS_TIMESTAMP,
+}
+
+UNITS = {
+    CONF_CELSIUS: TEMP_CELSIUS,
+    CONF_SECONDS: TIME_SECONDS,
+    CONF_KELVIN: TEMP_KELVIN,
+    CONF_BAR: PRESSURE_BAR,
+    CONF_PERCENT: PERCENTAGE_PERCENT,
+    CONF_ENERGY: ENERGY_KILO_WATT_HOUR,
+    CONF_VOLTAGE: VOLTAGE_VOLT,
+    CONF_HOURS: TIME_HOUR,
+    CONF_FLOW: FLOW_LITERS_PER_MINUTE,
+}
```

Comment:
Thinking about it again IMHO it would be easier to read if you move all of these to const.py and import them.